unittests/NodeAPI/child_process_mac.cpp

```diff
@@ -0,0 +1,141 @@
+#include "child_process.h"
+
+#include <fcntl.h>
+#include <spawn.h>
+#include <sys/wait.h>
+#include <unistd.h>
+#include <cassert>
+#include <cstdio>
+#include <cstring>
+#include <sstream>
+#include <string>
+#include <vector>
+
+// Verify the condition.
+// - If true, resume execution.
+// - If false, print a message to stderr and exit the app with exit code 1.
+#ifndef VerifyElseExit
+#define VerifyElseExit(condition)                                              \
+  do {                                                                         \
+    if (!(condition)) {                                                        \
+      ExitOnError(#condition, nullptr);                                        \
+    }                                                                          \
+  } while (false)
+#endif
+
+// Verify the condition.
+// - If true, resume execution.
+// - If false, destroy the passed `posix_spawn_file_actions_t* actions`, then
+// print a message to stderr and exit the app with exit code 1.
+#ifndef VerifyElseExitWithCleanup
+#define VerifyElseExitWithCleanup(condition, actions_ptr)                      \
+  do {                                                                         \
+    if (!(condition)) {                                                        \
+      ExitOnError(#condition, actions_ptr);                                    \
+    }                                                                          \
+  } while (false)
+#endif
```

Comment:
Wasn't sure whether to make an additional macro that effectively accepts `actions_ptr` as an optional argument, so kept it simple and explicit for now.